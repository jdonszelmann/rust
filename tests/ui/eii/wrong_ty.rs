//@ compile-flags: --crate-type rlib
#![feature(eii)]
#![feature(decl_macro)]
#![feature(rustc_attrs)]
#![feature(eii_internals)]

#[core::eii_macro_for(bar)]
#[rustc_builtin_macro(eii_macro)]
macro foo() {

}

unsafe extern "Rust" {
    safe fn bar(x: u64) -> u64;
}

#[foo]
fn other() -> u64 {
//~^ ERROR function `other` has a type that is incompatible with the declaration
    3
}

fn main() {
    bar(0);
}
