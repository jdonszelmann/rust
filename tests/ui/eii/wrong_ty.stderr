error[E0053]: function `other` has a type that is incompatible with the declaration
  --> $DIR/wrong_ty.rs:18:1
   |
LL | fn other() -> u64 {
   | ^^^^^^^^^^^^^^^^^ incorrect number of function parameters
   |
note: type in declaration
  --> $DIR/wrong_ty.rs:14:5
   |
LL |     safe fn bar(x: u64) -> u64;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: expected signature `fn(u64) -> _`
              found signature `fn() -> _`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0053`.
